test/component/instrumentations.compatibility.test.ts

```diff
@@ -0,0 +1,93 @@
+import 'jest-chain';
+import fs from 'fs';
+const rimraf = require('rimraf');
+const semver = require('semver');
+import { watchDir } from './helpers/fileListener';
+import { waitForChildProcess } from './helpers/helpers';
+import { instrumentationsVersionManager } from './helpers/InstrumentationsVersionManager';
+import { InstrumentationTest } from './instrumentations/InstrumentationTest';
+
+describe("'All Instrumentation's tests'", () => {
+  afterAll(() => {
+    const versions = instrumentationsVersionManager.getInstrumantaionsVersions();
+    Object.keys(versions).forEach((lib) => {
+      // updated supported versions file
+      if (!fs.existsSync(`${__dirname}/../../instrumentations/${lib}/tested_versions`)) {
+        fs.mkdirSync(`${__dirname}/../../instrumentations/${lib}/tested_versions`);
+      }
+      const versionStrings = versions[lib].unsupported
+        .map((v) => `!${v}`)
+        .concat(versions[lib].supported)
+        .sort((v1, v2) => semver.compare(v1.replace('!', ''), v2.replace('!', '')))
```

Comment:
Do you have "!" in the string? It's already in `unsupported` array.
If "!" doesn't exist then it should be added later.

Reply:
its sorting after adding it